### PR TITLE
chore: bump monorepo version to 1.12.0-rc1

### PR DESCRIPTION
**Pull request by jjyr**
[CHANGELOG](https://github.com/jjyr/godwoken/blob/deaa095c57edc0a599453ca003f4b0684a4f4971/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,28 @@ The format is based on [Keep a Changelog](https://keepachangelog.com).
 
 ## [Unreleased]
 
+## [v1.12.0-rc1] - 2023-02-08
+
+The `web3` and `web3-indexer` components have been added to the monorepo since this release, and we bumped the version from `v1.8.x` to `v1.12.x` to unify the version.
+
+We introduce a breaking change of the config file in [#946](https://github.com/godwokenrises/godwoken/pull/946). The consensus-related options are moved into option `consensus`, and we put the `testnet` and `mainnet` consensus builtin into the godwoken program. This change reduces the operation works of node maintainers.
```
Comment (Flouse) [inline CHANGELOG.md]:
```suggestion
We introduced a breaking change of the config file in [#946](https://github.com/godwokenrises/godwoken/pull/946). The consensus-related options are moved into option `consensus`, and we put the builtin `testnet` and `mainnet` consensus into the godwoken program. This change reduces the operation works of node maintainers.
```

**Review by Flouse — CHANGES_REQUESTED** [review]

**Review by Flouse — APPROVED** [review]

**Review by Simon-Tl — APPROVED** [review]

File: web3/package.json [hunk 2]
```diff
@@ -30,6 +30,6 @@
     "eslint": "^8.5.0",
     "prettier": "^2.2.1"
   },
-  "version": "1.11.0",
+  "version": "1.12.0-rc1",
```
Comment (RetricSu) [inline web3/package.json:33]: we should remove the special tag handling in web3 CI:
- web3-docker-publish.yml
- web3-indexer-docker-publish.yml

eg:

- https://github.com/godwokenrises/godwoken/blob/develop/.github/workflows/web3-docker-publish.yml#L69-L74
- https://github.com/godwokenrises/godwoken/blob/develop/.github/workflows/web3-indexer-docker-publish.yml#L9

File: web3/Dockerfile [hunk 3]
```diff
@@ -1,9 +1,9 @@
 
 FROM node:14-bullseye
 
-COPY . /godwoken/.
+COPY ./web3 /godwoken-web3
 
-WORKDIR /godwoken/web3
+WORKDIR /godwoken-web3
```
Comment (RetricSu) [inline web3/Dockerfile]:
```suggestion
```

File: web3/Dockerfile [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
 
 RUN yarn && yarn build
```
Comment (RetricSu) [inline web3/Dockerfile]:
```suggestion
RUN cd /godwoken-web3 && yarn && yarn build
```
see https://github.com/godwokenrises/godwoken/commit/1a3174548febd199a44ccd331ba4d64299676014#diff-6ec036556353d556083fa7858cef52cfece85508440e56c79c9c9d3476b96a35L5

**Review by RetricSu — APPROVED** [review]

**Pull request merged**